src/mesh/mesh-amr_loadbalance.cpp

```diff
@@ -644,9 +644,8 @@ bool Mesh::GatherCostListAndCheckBalance() {
 void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput *app_in,
                                            int ntot) {
   PARTHENON_INSTRUMENT
-  // kill any cached packs
-  mesh_data.PurgeNonBase();
-  mesh_data.Get()->ClearCaches();
+  // kill all old MeshData
+  mesh_data.clear();
```

Comment:
does this have implications? I guess we always rebuild these at the beginning of a step...

Reply:
I am pretty sure this is safe, since the base MeshData container is rebuilt at the end of this routine and I think `mesh_data` isn't referenced anywhere in between. Previously, everything but the base container was purged but then the base container was rebuilt, so I think everything could have been safely purged. Definitely having another set of eyes on this to make sure I didn't miss some subtlety would be helpful.